runtime/mock/src/circuit_config.rs

```diff
@@ -1,18 +1,100 @@
 use crate::*;
 
-use frame_support::{parameter_types, traits::ConstU32, PalletId};
+use frame_support::{parameter_types, traits::ConstU32, weights::Weight, PalletId};
 use pallet_grandpa_finality_verifier::bridges::runtime as bp_runtime;
+
 use sp_core::H256;
+use sp_runtime::Perbill;
+
+use sp_runtime::traits::One;
 
 impl t3rn_primitives::EscrowTrait<Runtime> for Runtime {
     type Currency = Balances;
     type Time = Timestamp;
 }
 
+pub struct GlobalOnInitQueues;
+
+impl pallet_clock::traits::OnHookQueues<Runtime> for GlobalOnInitQueues {
+    fn process(n: BlockNumber, on_init_weight_limit: Weight) -> Weight {
+        let mut weights_consumed = vec![];
+        // Iterate over all pre-init hooks implemented by pallets and return aggregated weight
+        weights_consumed.push(Circuit::process_signal_queue(
+            n,
+            BlockNumber::one(),
+            Perbill::from_percent(15) * on_init_weight_limit,
+        ));
+        log::debug!(
+            "Circuit::process_signal_queue consumed: {:?}",
+            weights_consumed
+                .last()
+                .expect("Circuit::process_signal_queue consumed weight")
+        );
+        weights_consumed.push(Circuit::process_xtx_tick_queue(
+            n,
+            BlockNumber::one(),
+            Perbill::from_percent(35) * on_init_weight_limit,
+        ));
+        log::debug!(
+            "Circuit::process_xtx_tick_queue consumed: {:?}",
+            weights_consumed
+                .last()
+                .expect("Circuit::process_xtx_tick_queue consumed weight")
+        );
+        weights_consumed.push(Circuit::process_revert_xtx_queue(
+            n,
+            10u32,
+            Perbill::from_percent(35) * on_init_weight_limit,
+        ));
+        log::debug!(
+            "Circuit::process_revert_xtx_queue consumed: {:?}",
+            weights_consumed
+                .last()
+                .expect("Circuit::process_revert_xtx_queue consumed weight")
+        );
+        weights_consumed.push(Clock::check_bump_round(
+            n,
+            BlockNumber::one(),
+            Perbill::from_percent(5) * on_init_weight_limit,
+        ));
+        log::debug!(
+            "Clock::check_bump_round consumed: {:?}",
+            weights_consumed
+                .last()
+                .expect("Clock::check_bump_round consumed weight")
+        );
+        weights_consumed.push(Clock::calculate_claimable_for_round(
+            n,
+            BlockNumber::one(),
+            Perbill::from_percent(10) * on_init_weight_limit,
```

Comment:
I would suggest initially adding all these numbers at the top to ensure they are <= 100 as i'd expect some interesting behaviour if it's more than 100.

Reply:
done